text/2018-08-29-destroy-range.md

```diff
@@ -0,0 +1,50 @@
+# Summary
+
+Support RPC `DestroyRange`. This call is on the whole TiKV rather than a certain region. When it is invoked, TiKV will use `delete_files_in_range` to quickly free a large amount of space and delete all keys in the range. Raft layer will be bypassed, and the range can cover across multiple regions. The invoker should promise that after invoking `DestroyRange`, the range will never be accessed again. That is to say, the range is permanently scrapped.
+
+This interface is only designed for TiDB.
+
+# Motivation
+
+Currently, when TiDB deletes a range, it will invoke `DeleteRange` on all affected regions respectively. Though TiKV will call `delete_files_in_range` to quickly free up the space for each region, the SST files that span over multiple regions can't be removed immediately. As a result, the disk space can't be recycled in time. Another problem is, the process to delete all regions respectively is too slow.
+
+This issue has troubled some of our users. It should be fixed as soon as possible.
+
+# Detailed design
+
+1. Add protobuf for `DestroyRange` to project kvrpcpb.
+    ```protobuf
+    message DestroyRangeRequest {
```

Comment:
It's a highly dangerous operation. Do we need to include some mechanism to avoid potential accidental usage?

Reply:
maybe name UnsafeDestroyRangeRequest?

Reply:
@zhangjinpeng1987 What do you think?